### PR TITLE
Yet more TSDoc

### PR DESCRIPTION
**Pull request by jonathan-codaio**
This is basically all of the rest of the reachable entities. After this, I can send a followup to rework tsdoc_progress.ts as a unittest that enforces that all reachable entities have documentation.

This isn't quite the end of it though, all of the entities that typedoc is currently throwing warnings about will need to be exported and documented too. I'll work on those next.

PTAL @ekoleda-codaio @coda/packs

### CODE REVIEW

File: schema.ts [hunk 1]
```diff
@@ -537,40 +537,151 @@ export interface ArraySchema<T extends Schema = Schema> extends BaseSchema {
...
+   *   },
+   * });
+   * ```
+   *
+   * This tells Coda to transform your formula's return value, creating a field "durationSeconds"
+   * whose value comes another field called "duration".
+   */
   fromKey?: string;
+  /**
+   * When true, indicates that an object return value for a formula that has this schema must
```
Comment (ekoleda-codaio) [inline schema.ts:578]: Does this just add build-time checks? What happens though at runtime if you don't return it?

File: schema.ts [hunk 2]
```diff
@@ -537,40 +537,151 @@ export interface ArraySchema<T extends Schema = Schema> extends BaseSchema {
...
 };
 
+/** @hidden */
 export type GenericObjectSchema = ObjectSchema<string, string>;
 
+// TODO(jonathan, ekoleda): Link to a guide about identities and references.
+/**
+ * An identifier for a schema, allowing other schemas to reference it.
+ *
+ * You may optionally specify an {@link identity} when defining an object schema.
```
Comment (ekoleda-codaio) [inline schema.ts:600]: Some of these links aren't working correctly:

https://coda-packs-sdk--1538.com.readthedocs.build/en/1538/reference/sdk/interfaces/IdentityDefinition/

Should they have been prefixed with a type name?

File: schema.ts [hunk 3]
```diff
@@ -537,40 +537,151 @@ export interface ArraySchema<T extends Schema = Schema> extends BaseSchema {
...
 export interface IdentityDefinition {
+  /**
+   * The name of this entity. This is an arbitrary name but should be unique within your pack.
+   * For example, if you are defining a schema that represents a user object, "User" would be a good identity name.
+   */
   name: string;
+  // TODO(jonathan): Inject the dynamicUrl at runtime like we do the packId so that pack makers
+  // don't have to it themselves.
+  /**
+   * The dynamic URL, this is a schema for a dynamic sync table. When returning a schema from the {@link getSchema}
```
Comment (ekoleda-codaio) [inline schema.ts:619]: Are you missing an "if" here? ", if this is a ..."

File: types.ts [hunk 4]
```diff
@@ -210,7 +212,25 @@ export enum PostSetupType {
...
 
 interface BaseAuthentication {
+  /**
+   * A function that is called when a user sets up a new account, that returns a name for
+   * the account to label that account in the UI. The users credentials are applied to any
+   * fetcher requests that this function makes. Typically, this function makes an API call
+   * to an API's "who am I" endpoint and returns a username.
+   *
+   * If omitted, or if the function returns an empty value, the account will be labeled
+   * with the creating user's Coda email address.
```
Comment (ekoleda-codaio) [inline types.ts:222]: In my experience in the UI it's labeled with the user's username, not email address.

**Review by ekoleda-codaio — CHANGES_REQUESTED** [review]

**Comment by jonathan-codaio** [comment]
Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.

File: schema.ts (hunk 1 above)
Comment (ekoleda-codaio) [reply schema.ts:578]: The key question I'm getting at is "what happens if I screw up and it's empty?" It sounds like the references that rely on the field won't work, but in general your Pack won't blow up.

File: schema.ts (hunk 2 above)
Comment (ekoleda-codaio) [reply schema.ts]: Just checked the same preview link and those aren't working either.

**Comment by ekoleda-codaio** [comment]
> Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.

Looks like the links are still broken, but once those are fixed feel free to merge.

**Comment by jonathan-codaio** [comment]
> > Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.
> 
> Looks like the links are still broken, but once those are fixed feel free to merge.

Cool, will look into these, but mind approving so I can try to merge if you've already signed off for the night?

**Comment by jonathan-codaio** [comment]
> > > Mind if I merge this and make followups? I'm working on the final cleanup and trying to avoid major rebase conflicts before I get too far.
> > 
> > 
> > Looks like the links are still broken, but once those are fixed feel free to merge.
> 
> Cool, will look into these, but mind approving so I can try to merge if you've already signed off for the night?

Ok these are fixed. Required adding more exports, which in turn required adding more tsdoc, which was pulling on the thread that I was trying to address in my follow-on PR to start dealing with typedoc warnings by adding new exports and entrypoints. Each new import brings in a bunch of new identifiers with it so fixing any single one of these can be deep rabbithole, now that we have doc comment enforcement turned on.

**Review by patrick-codaio — APPROVED** [review]

**Pull request merged**